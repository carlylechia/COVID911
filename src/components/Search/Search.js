import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { filterCountries } from '../../Redux/countries/countries';

const Search = () => {
  const notDisplay = useLocation().pathname.replace('/', '');
  const input = useSelector((state) => state.regionsReducer.input);
  const dispatch = useDispatch();
  return (
    <input
      className={`${notDisplay.length ? 'none' : 'block'}`}
      style={{
        borderWidth: '2px', borderRadius: '1rem', padding: '5px', fontSize: '1rem', textAlign: 'center', margin: '2rem auto',
      }}
      type="text"
      value={input}
      placeholder="Search Country"
      onChange={(e) => {
        dispatch(filterCountries(e.target.value));
      }}
    />
  );
};
export default Search;
