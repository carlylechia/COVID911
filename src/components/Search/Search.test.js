import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/createStore';
import Search from './Search';

describe('Search component test', () => {
  test('Search snapshot', () => {
    const search = render(
      <Router>
        <Provider store={store}>
          <Search />
        </Provider>
      </Router>,
    );
    expect(search).toMatchSnapshot();
  });
});
