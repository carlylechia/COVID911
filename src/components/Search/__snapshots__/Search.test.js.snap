// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component test Search snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="block"
        placeholder="Search Country"
        style="border-width: 2px; border-radius: 1rem; padding: 5px; font-size: 1rem; text-align: center; margin: 2rem auto;"
        type="text"
        value=""
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="block"
      placeholder="Search Country"
      style="border-width: 2px; border-radius: 1rem; padding: 5px; font-size: 1rem; text-align: center; margin: 2rem auto;"
      type="text"
      value=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
